dali/pipeline/util/bounding_box.h

```diff
@@ -49,24 +50,26 @@ class BoundingBox {
     swap(lhs.area_, rhs.area_);
   }
 
-  static BoundingBox FromLtrb(const float* data) {
-    return FromLtrb(data[0], data[1], data[2], data[3]);
+  static BoundingBox FromLtrb(const float* data, bool check_input = true) {
```

Comment:
Unclear why we need a check_input flag here. I get the idea of overcoming the limitation of 0-1 (if that is what you're after), but in this case I don't think it's the right approach as it introduces a little gotcha that can make it problematic to debug.

If you would like to introduce something like that, my proposition would be to make this api more generic but without sidestepping. A proposition could be to explicitly pass the bounds as a pair which by default would be {std::numeric_limits<float>::min(), std::numeric_limits<float>::max()}. In this case, any value for our bboxes would validate, while keeping the checks consistent. If you'd like to have 0-1 bounds for values in BoundingBox you could call it BoundingBox(0,0,0,0, {0,1}).

If this is too much (may be), we can drop the 0-1 boundary restriction altogether, but the constrants l <= r and t <=b must be in place because otherwise the box is malformed. As it stands, the boolean flag is creating a problem but we should aspire to go for the fix and not the workaround for this kind of types that are meant to be used accross operators.